### DISCUSSION TITLE
Additional shuffle options: Shuffle by Album and Shuffle by Artist

### DISCUSSION THREAD

**Discussion by swiewiora** [music-assistant/support#3118] [Feature requests and ideas]
I would like to have a possibility to shuffle playback of the playlist not by every track but after playing complete album or all songs from a single artist. For reference, this functionality exists in a Rocket Player for Android.

**Comment by OzGav** [comment]
I’m not sure what you are asking for. If the queue reaches the end what is the point of enabling shuffle at that point? If you are then adding more tracks to the queue manually then you can set the shuffle option on yourself via the UI or with the HA action

**Comment by flvinny521** [comment]
I came here to make a similar request. My desire is to have MA shuffle my playback queue by albums instead of tracks. In other words, if I load 10 albums into my queue, I don't want to hear all the corresponding tracks shuffled, but to hear each album from start to finish in a shuffled order.

I assume the original poster was also looking for the ability to listen to all tracks by a specific artist, before moving on to all tracks by a different artist, where the artists are shuffled and not played in alphabetical order.

I have just installed Music Assistant for the first time, so apologies if there is already a way to accomplish this.

**Reply by OzGav** [reply]
If you use HA you could do this

```
action: music_assistant.get_library
data:
  config_entry_id: 01K4MWK4Y5ZFHXVTDXPZCDF8N6
  media_type: album
  search: AC/DC
  order_by: random
```

Put those in a response variable and then queue up the albums. Related example here https://www.music-assistant.io/faq/how-to/#play-a-random-item

**Reply by flvinny521** [reply]
Thanks for that info. I am just starting out with Home Assistant but I'll play around with this to see if I can figure it out. Any tips on modifying that code above if I wanted to shuffle all the albums in my library instead of searching for one specific artist? I'm using the Subsonic provider connected to my Navidrome server, if it makes any difference.

**Reply by OzGav** [reply]
Just omit the search parameter
